sos/plugins/dnf.py

```diff
@@ -60,4 +66,17 @@ def setup(self):
         if self.get_option("history"):
             self.add_cmd_output("dnf history")
 
+        if self.get_option("history-info"):
+            history = self.call_ext_prog("dnf history")
```

Comment:
Ran each of these 3 times, on two machines - rhel7 and fedora24. Time is the best of 3. The fedora machine has 57 entries in dnf history, the rhel7 one has 28 in yum history.

time sosreport --batch
fedora24 22s
rhel7 42s

time sosreport -k yum.yum-history-info=on --batch
rhel7 52s

time sosreport -k dnf.history-info=on -k dnf.history=on --batch
fedora24 44s

So around 10 seconds more on the rhel7 machine, 22 seconds more on the fedora24 one. Since it seems somewhat proportional to the number of transactions, I'd say it adds around half a second per transaction. Very unscientific, I know.

Didn't try just adding yum/dnf history (without info per transaction), but when running them manually they take around 0.3s on rhel7, 0.6s on fedora24 (or probably around 0.01 seconds per entry).

Also note that at least on my rhel7 machine, the output of 'yum history' is cut in the middle:
...
    11 | downgrade http://bob.eng | 2016-06-14 11:49 | Downgrade      |    1   
    10 | install http://bob.eng.l | 2016-06-14 11:43 | Update         |    1   
     9 |                          | 2016-06-14 11:33 | I, U           |   28 EE
history list

but e.g. 'yum history info 7' does show info. Might be a problem in yum, but I'd rather collect all the info anyway.

So bottom line, I'd say the time added is significant, but IMO acceptable.